--- boost_1_73_0/tools/build/src/engine/build.sh.orig   2020-11-18 15:09:52.222115087 +0100
+++ boost_1_73_0/tools/build/src/engine/build.sh        2020-11-18 15:18:59.500663256 +0100
@@ -81,7 +81,7 @@
         case $1 in
             gcc) ( ${CXX:=g++} -x c++ -std=c++11 check_cxx11.cpp && rm -f a.out ) 1>/dev/null 2>/dev/null ;;
             intel-darwin) ( ${CXX:=icc} -xc++ check_cxx11.cpp && rm -f a.out ) 1>/dev/null 2>/dev/null ;;
-            intel-linux) ( ${CXX:=icc} -xc++ check_cxx11.cpp && rm -f a.out ) 1>/dev/null 2>/dev/null ;;
+            intel-linux) ( ${CXX:=icpc} -x c++ -std=c++11 check_cxx11.cpp && rm -f a.out ) 1>/dev/null 2>/dev/null ;;
             vacpp) ( ${CXX:=xlC_r} check_cxx11.cpp && rm -f a.out ) 1>/dev/null 2>/dev/null ;;
             xlcpp) ( ${CXX:=xlC_r} check_cxx11.cpp && rm -f a.out ) 1>/dev/null 2>/dev/null ;;
             como) ( ${CXX:=como} check_cxx11.cpp && rm -f a.out ) 1>/dev/null 2>/dev/null ;;
@@ -124,7 +124,10 @@
     elif test_uname FreeBSD && test_path freebsd-version && test_path clang++ && test_cxx11 clang ; then B2_TOOLSET=clang
     elif test_path g++ && test_cxx11 gcc ; then B2_TOOLSET=gcc
     elif test_path clang++ && test_cxx11 clang ; then B2_TOOLSET=clang
-    elif test_path icc && test_cxx11 intel-linux ; then B2_TOOLSET=intel-linux
+    elif test_path icpc && test_cxx11 intel-linux ; then B2_TOOLSET=intel-linux
+    elif test -r /opt/intel/inteloneapi/setvars.sh && test_cxx11 intel-linux ; then
+        B2_TOOLSET=intel-linux
+        B2_TOOLSET_ROOT=/opt/intel/inteloneapi
     elif test -r /opt/intel/cc/9.0/bin/iccvars.sh && test_cxx11 intel-linux ; then
         B2_TOOLSET=intel-linux
         B2_TOOLSET_ROOT=/opt/intel/cc/9.0
@@ -247,18 +250,19 @@
     ;;

     intel-linux)
-        CXX=${CXX:=icc}
+        CXX=${CXX:=icpc}
         CXX_VERSION_OPT=${CXX_VERSION_OPT:=--version}
-        test_path ${CXX} >/dev/null 2>&1
-        if test $? ; then
+       if test_path ${CXX} ; then
             echo "Found ${CXX} in environment"
-            B2_TOOLSET_ROOT=`echo ${CXX}| sed -e 's/bin.*\/icc//'`
+            B2_TOOLSET_ROOT=`echo ${CXX}| sed -e 's/bin.*\/icpc//'`
             # probably the most widespread
             ARCH=intel64
         else
             echo "No intel compiler in current path"
-            echo "Look in a few old place for legacy reason"
-            if test -r /opt/intel/cc/9.0/bin/iccvars.sh ; then
+            echo "Look in a few common places just in case"
+            if test -r /opt/intel/inteloneapi/setvars.sh ; then
+                B2_TOOLSET_ROOT=/opt/intel/inteloneapi
+            elif test -r /opt/intel/cc/9.0/bin/iccvars.sh ; then
                 B2_TOOLSET_ROOT=/opt/intel/cc/9.0/
             elif test -r /opt/intel_cc_80/bin/iccvars.sh ; then
                 B2_TOOLSET_ROOT=/opt/intel_cc_80/
@@ -270,9 +274,11 @@
                 B2_TOOLSET_ROOT=/opt/intel/compiler50/ia32/
             fi
         fi
-        if test -r ${B2_TOOLSET_ROOT}bin/iccvars.sh ; then
+        if test -r ${B2_TOOLSET_ROOT}/setvars.sh ; then
+            . ${B2_TOOLSET_ROOT}/setvars.sh
+        elif test -r ${B2_TOOLSET_ROOT}bin/iccvars.sh ; then
             # iccvars does not change LD_RUN_PATH. We adjust LD_RUN_PATH here in
-            # order not to have to rely on ld.so.conf knowing the icc library
+            # order not to have to rely on ld.so.conf knowing the icpc library
             # directory. We do this before running iccvars.sh in order to allow a
             # user to add modifications to LD_RUN_PATH in iccvars.sh.
             if test -z "${LD_RUN_PATH}"; then
@@ -283,7 +289,7 @@
             export LD_RUN_PATH
             . ${B2_TOOLSET_ROOT}bin/iccvars.sh $ARCH
         fi
-        B2_CXX="${CXX} -xc++"
+        B2_CXX="${CXX} -x c++ -std=c++11"
         B2_CXXFLAGS_RELEASE="-O3 -s"
         B2_CXXFLAGS_DEBUG="-O0 -g -p"
     ;;
